src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -337,7 +338,15 @@ private void LogPrimaryKeys()
                 {
                     // MsSql types like date,smalldatetime,datetime,datetime2 are mapped to the same .NET type of DateTime.
                     // Thus to determine the actual dbtype, we use the underlying MsSql type instead of the .NET type.
-                    paramDefinition.DbType = ResolveDbType((string)row["DATA_TYPE"]);
+                    DbType dbType;
+                    if (TryResolveDbType(sqlType, out dbType))
+                    {
+                        paramDefinition.DbType = dbType;
+                    }
+                    else
+                    {
+                        _logger.LogWarning("Could not determine DbType for SqlDb type of {sqlType}", sqlType);
```

Comment:
What are the implications of this warning from the user's perspective:
- does the user need to act?
- will the user expect to see other errors because of this?
- will the user rarely, if ever, see this warning printed?

Reply:
Again, this would only be hit when a new data type is added in MsSql which maps to .NET type of `DateTime`. The user cannot do anything regarding this. This would involve a code change to add a mapping for the new data type.